packages/forms/frontend/sirius-components-forms/src/propertysections/TreePropertySection.tsx

```diff
@@ -45,26 +146,107 @@ const TreeItem = ({ node, nodes }: TreeItemProps) => {
       setSelection({ entries: [newSelection] });
     }
   };
+
   const label = (
-    <div className={styles.label} onClick={handleClick}>
+    <div className={classes.label} onClick={handleClick}>
+      {node.checkable ? (
+        <Checkbox
+          name={node.label}
+          checked={node.value}
+          onChange={onChange}
+          onClick={(event) => event.stopPropagation()}
+          data-testid={node.label}
+          disabled={readOnly}
```

Comment:
Only the border of the checkbox is gray when the whole tree is disabled. I think the whole Tree should be grayed (checkbox, label, but not the title). I don't know for the icons, may be not.
If you test a read only Checkbox or a read only Textfield, you will see that the whole widget is grayed (except for the title of the widget).
Could you update this please?